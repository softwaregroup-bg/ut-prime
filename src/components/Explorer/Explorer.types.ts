import React from 'react';
import Joi from 'joi';
import { DataTableProps, DataViewProps } from '../prime';

import type {Schema, Cards, Editors, Dropdowns, WidgetReference, Layouts} from '../types';

interface Customization {
    schema?: Schema,
    card?: Cards,
    layout?: Layouts
}

export interface Props {
    /**
     * Name of the key field in the result set.
     */
    keyField?: string;
    /**
     * Name of the property, in which the result set is returned.
     */
    resultSet?: string;
    /**
     * Schema defining the properties in the result set.
     */
    schema: Schema;
    /**
     * Array of property names to show as columns.
     * @deprecated use cards.xxx.widgets instead
     */
    columns?: WidgetReference[];
    /**
     * Data fetching async function.
     */
    fetch: (params: {
        [resultSet: string]: object,
        orderBy?: {
            field: string,
            dir: string
        }[],
        paging?: {
            pageSize: number,
            pageNumber: number
        }
    }) => Promise<{
        pagination?: {
            recordsTotal: number
        },
        [data: string]: unknown
    }>;
    fetchTransform?: (params: unknown) => unknown,
    subscribe?: (callback: (rows: [] | Record<string, object>) => void) => () => void;
    onDropdown?: (params: string[]) => Promise<Dropdowns>,
    className?: string;
    /**
     * Fields to show in the details pane.
     */
    details?: {
        page: string,
        params?: unknown
    };
    toolbar?: false | WidgetReference[];
    filter?: Record<string, unknown>;
    params?: Record<string, unknown>;
    index?: object;
    showFilter?: boolean;
    design?: boolean;
    pageSize?: number;
    table?: DataTableProps;
    editors?: Editors;
    view?: DataViewProps;
    cards?: Cards;
    layout?: string;
    layouts?: Layouts;
    methods?: object;
    name?: string;
    hidden?: boolean;
    refresh?: boolean;
    customization?: Customization,
    paramsCard?: string,
    onFieldChange?: string,
    value?: {selected?: unknown[]},
    onChange?: (params: object) => void,
    onCustomization?: (customization: {component: {componentId: string, componentConfig: Customization}}) => Promise<object>;
    fetchValidation?: Joi.Schema
}

export type ComponentProps = React.FC<Props>
