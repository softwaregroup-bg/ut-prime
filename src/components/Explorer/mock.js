const startDate = new Date(2022, 5, 22);

if (!window?.navigator?.userAgent?.match?.(/Chromatic/)) {
    startDate.setTime(Date.now());
    startDate.setHours(-36, 0, 0, 0);
}

const filteredItems = (filterBy) => {
    return Object.entries(filterBy).reduce((items, [key, value]) => {
        return items.filter(i => {
            if (Array.isArray(value) && (value[0] instanceof Date || value[1] instanceof Date)) {
                return (value[0] == null || value[0] <= i[key]) && (value[1] == null || i[key] < value[1]);
            }
            return value == null || i[key] === value || (typeof value === 'string' && i[key].toString().toLowerCase().includes(value.toLowerCase()));
        });
    }, [...Array(55).keys()].map(number => ({
        id: number,
        name: `Item ${number}`,
        date: new Date(startDate.getTime() + 1000 * 60 * 60 * 24 * number),
        dateTime: new Date(startDate.getTime() + 1000 * 60 * 60 * number),
        time: new Date(1000 * 60 * 10 * number),
        size: number * 10
    })));
};

const compare = ({field, dir, smaller = {ASC: -1, DESC: 1}[dir]}) => function compareMock(a, b) {
    if (a[field] < b[field]) return smaller;
    if (a[field] > b[field]) return -smaller;
    return 0;
};

export const fetchItems = filters => {
    const items = filteredItems(filters.items);
    if (Array.isArray(filters.orderBy) && filters.orderBy.length) items.sort(compare(filters.orderBy[0]));
    return Promise.resolve({
        items: items.slice((filters.paging.pageNumber - 1) * filters.paging.pageSize, filters.paging.pageNumber * filters.paging.pageSize),
        pagination: {
            recordsTotal: items.length
        }
    });
};

export const updateItems = update => {
    let size = 0;
    // eslint-disable-next-line no-process-env
    const interval = (process.env.NODE_ENV === 'production') ? 10000 : 1000;
    const handle = setInterval(() => {
        size++;
        update({0: {size}});
    }, interval);
    return () => clearInterval(handle);
};
