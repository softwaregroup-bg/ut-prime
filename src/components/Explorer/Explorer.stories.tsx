import React from 'react';
import type { Story, Meta } from '@storybook/react';
import {Toast} from '../prime';

import page from './README.mdx';
import Explorer from './index';
import {fetchItems, updateItems} from './mock';
import decorators from '../test/decorator';

const meta: Meta = {
    title: 'Explorer',
    component: Explorer,
    parameters: {docs: {page}},
    decorators,
    args: {
        state: {}
    }
};
export default meta;

const Template: Story<{
    createPermission?: string,
    editPermission?: string,
    deletePermission?: string,
    details?: object,
    children?: React.ReactNode,
    columns?: string[]
}> = ({
    createPermission,
    editPermission,
    deletePermission,
    details,
    children,
    columns = ['name', 'size'],
    ...props
}) => {
    const toast = React.useRef(null);
    const show = action => data => toast.current.show({
        severity: 'success',
        summary: 'Submit',
        detail: <pre>{JSON.stringify({action, data}, null, 2)}</pre>
    });
    return (
        <>
            <Toast ref={toast} />
            <div style={{height: 'fit-content', display: 'flex', flexDirection: 'column'}}>
                <Explorer
                    fetch={fetchItems}
                    keyField='id'
                    resultSet='items'
                    schema={{
                        properties: {
                            id: {
                                action: show('action')
                            },
                            name: {
                                action: show('action'),
                                title: 'Name',
                                filter: true,
                                sort: true
                            },
                            date: {
                                title: 'Date',
                                filter: true,
                                sort: true,
                                format: 'date'
                            },
                            time: {
                                title: 'Time',
                                filter: true,
                                sort: true,
                                format: 'time'
                            },
                            dateTime: {
                                title: 'Date and time',
                                filter: true,
                                sort: true,
                                format: 'date-time'
                            },
                            size: {
                                title: 'Size',
                                filter: true,
                                sort: true
                            }
                        }
                    }}
                    columns = {columns}
                    subscribe={updateItems}
                    details={details}
                    filter={{}}
                    toolbar={[{
                        title: 'Create',
                        permission: createPermission,
                        action: () => {}
                    }, {
                        title: 'Edit',
                        permission: editPermission,
                        enabled: 'current',
                        action: show('edit')
                    }, {
                        title: 'Delete',
                        permission: deletePermission,
                        enabled: 'selected',
                        action: show('delete')
                    }]}
                    {...props}
                >
                    {children}
                </Explorer>
            </div>
        </>
    );
};

export const Basic = Template.bind({});
Basic.args = {};

export const Children = Template.bind({});
Children.args = {
    ...Basic.args,
    children: <div>Navigation component</div>
};

export const Details = Template.bind({});
Details.args = {
    ...Children.args,
    details: {
        name: 'Name'
    }
};

export const ActionPermissions = Template.bind({});
ActionPermissions.args = {
    ...Details.args,
    createPermission: 'forbidden',
    editPermission: 'granted',
    deletePermission: 'forbidden'
};

export const DateTimeFilter = Template.bind({});
DateTimeFilter.args = {
    ...Basic.args,
    columns: ['date', 'time', 'dateTime']
};

export const Grid = Template.bind({});
Grid.args = {
    ...Basic.args,
    cards: {
        grid: {
            type: 'grid',
            widgets: [{
                name: 'name', type: 'label'
            }, {
                name: 'size', type: 'label'
            }]
        }
    },
    pageSize: 36,
    layout: ['grid']
};

export const GridFlex = Template.bind({});
GridFlex.args = {
    ...Basic.args,
    cards: {
        grid: {
            type: 'grid',
            className: 'col-6 md:col-3',
            classes: {
                default: {
                    root: 'grid m-0',
                    widget: 'text-center col-4 mb-0'
                }
            },
            widgets: [{
                type: 'icon', title: 'pi-paperclip'
            }, {
                name: 'name', type: 'label'
            }, {
                name: 'size', type: 'label'
            }]
        }
    },
    pageSize: 24,
    layout: ['grid']
};
