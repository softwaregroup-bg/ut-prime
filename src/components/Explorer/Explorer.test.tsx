import React from 'react';
import {waitFor} from '@testing-library/react';

import { render } from '../test';
import Explorer from './index';
import {fetchItems} from './mock';

describe('<Explorer />', () => {
    it('render equals snapshot', async() => {
        const { getByTestId } = render(<Explorer
            fetch={fetchItems}
            keyField='id'
            resultSet='items'
            name='items'
            schema={{
                properties: {
                    name: {
                        title: 'Name',
                        filter: true,
                        sort: true
                    },
                    size: {
                        title: 'Size',
                        filter: true,
                        sort: true
                    }
                }
            }}
            cards = {{
                browse: {
                    widgets: ['name', 'size']
                }
            }}
            details={{
                page: 'details'
            }}
        />);
        await waitFor(() => expect(getByTestId('ut-front-test').querySelector('tr.p-selectable-row')).toBeTruthy());
        expect(getByTestId('ut-front-test')).toMatchSnapshot();
    });
});
