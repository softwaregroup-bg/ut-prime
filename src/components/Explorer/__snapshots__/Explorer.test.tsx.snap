// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Explorer /> render equals snapshot 1`] = `
<div
  data-testid="ut-front-test"
>
  <div
    class="flex flex-column explorer-0-2-2"
  >
    <div
      class="p-toast p-component p-toast-top-right"
    >
      <div />
    </div>
    <div
      class="p-toolbar p-component border-none p-2 flex-nowrap"
      role="toolbar"
      style="background: none;"
    >
      <div
        class="p-toolbar-group-left"
      />
      <div
        class="p-toolbar-group-right"
      >
        <button
          class="p-button p-component mr-2 ml-2 p-button-icon-only"
          data-testid="items-refreshButton"
        >
          <span
            class="p-button-icon p-c pi pi-search"
          />
          <span
            class="p-button-label p-c"
          >
             
          </span>
        </button>
        <button
          class="p-button p-component mr-2 p-button-icon-only"
          data-testid="items-details-toggleButton"
        >
          <span
            class="p-button-icon p-c pi pi-bars"
          />
          <span
            class="p-button-label p-c"
          >
             
          </span>
        </button>
        <button
          aria-label="design"
          class="p-button p-component p-button-icon-only"
          data-testid="itemsCustomization"
        >
          <span
            class="p-button-icon p-c pi pi-cog"
          />
          <span
            class="p-button-label p-c"
          >
             
          </span>
        </button>
      </div>
    </div>
    <div
      class="flex"
    >
      <div
        class="flex-grow-1"
        style="max-height: 768px;"
      >
        <div>
          <div
            class="p-splitter p-component p-splitter-horizontal"
            style="flex-grow: 1; height: 768px;"
          >
            <div
              class="p-splitter-panel"
              style="flex-basis: calc(90% - 4px); height: 768px;"
            >
              <div
                class="relative"
                style="height: 768px;"
              >
                <div
                  class="p-datatable p-component p-datatable-selectable p-datatable-auto-layout p-datatable-scrollable p-datatable-scrollable-vertical p-datatable-flex-scrollable p-datatable-responsive-stack"
                  data-scrollselectors=".p-datatable-wrapper"
                  pr_id_1=""
                >
                  <div
                    class="p-datatable-wrapper"
                    style="max-height: flex;"
                  >
                    <table
                      class="p-datatable-table"
                      role="table"
                    >
                      <thead
                        class="p-datatable-thead"
                      >
                        <tr
                          role="row"
                        >
                          <th
                            class="flex-grow-0 p-selection-column"
                            role="columnheader"
                          >
                            <div
                              class="p-column-header-content"
                            >
                              <span
                                class="p-column-title"
                              />
                            </div>
                          </th>
                          <th
                            aria-sort="none"
                            class="p-sortable-column"
                            role="columnheader"
                            tabindex="0"
                          >
                            <div
                              class="p-column-header-content"
                            >
                              <span
                                class="p-column-title"
                              >
                                <span
                                  data-testid="nameTitle"
                                >
                                  Name
                                </span>
                              </span>
                              <span
                                class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                              />
                            </div>
                          </th>
                          <th
                            aria-sort="none"
                            class="p-sortable-column"
                            role="columnheader"
                            tabindex="0"
                          >
                            <div
                              class="p-column-header-content"
                            >
                              <span
                                class="p-column-title"
                              >
                                <span
                                  data-testid="sizeTitle"
                                >
                                  Size
                                </span>
                              </span>
                              <span
                                class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                              />
                            </div>
                          </th>
                        </tr>
                        <tr
                          role="row"
                        >
                          <th
                            class="p-filter-column flex-grow-0"
                          >
                            <div
                              class="p-checkbox p-component"
                            >
                              <div
                                class="p-checkbox-box p-component"
                                role="checkbox"
                                tabindex="0"
                              >
                                <span
                                  class="p-checkbox-icon"
                                />
                              </div>
                            </div>
                          </th>
                          <th
                            class="p-filter-column"
                          >
                            <div
                              class="p-column-filter p-fluid p-column-filter-row"
                            >
                              <div
                                class="p-fluid p-column-filter-element"
                              >
                                <input
                                  class="p-inputtext p-component"
                                  name="nameFilter"
                                  value=""
                                />
                              </div>
                              <button
                                aria-label="Clear"
                                class="p-column-filter-clear-button p-link p-hidden-space"
                                type="button"
                              >
                                <span
                                  aria-hidden="true"
                                  class="pi pi-filter-slash"
                                />
                              </button>
                            </div>
                          </th>
                          <th
                            class="p-filter-column"
                          >
                            <div
                              class="p-column-filter p-fluid p-column-filter-row"
                            >
                              <div
                                class="p-fluid p-column-filter-element"
                              >
                                <input
                                  class="p-inputtext p-component"
                                  name="sizeFilter"
                                  value=""
                                />
                              </div>
                              <button
                                aria-label="Clear"
                                class="p-column-filter-clear-button p-link p-hidden-space"
                                type="button"
                              >
                                <span
                                  aria-hidden="true"
                                  class="pi pi-filter-slash"
                                />
                              </button>
                            </div>
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="p-datatable-tbody"
                      >
                        <tr
                          class="p-selectable-row"
                          role="row"
                        >
                          <td
                            class="flex-grow-0 p-selection-column"
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            />
                            <div
                              class="p-checkbox p-component"
                            >
                              <div
                                aria-checked="false"
                                aria-label="Select 0"
                                class="p-checkbox-box p-component"
                                role="checkbox"
                                tabindex="0"
                              >
                                <span
                                  class="p-checkbox-icon"
                                />
                              </div>
                            </div>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="nameTitle"
                              >
                                Name
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              Item 0
                            </span>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="sizeTitle"
                              >
                                Size
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              0
                            </span>
                          </td>
                        </tr>
                        <tr
                          class="p-selectable-row p-row-odd"
                          role="row"
                        >
                          <td
                            class="flex-grow-0 p-selection-column"
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            />
                            <div
                              class="p-checkbox p-component"
                            >
                              <div
                                aria-checked="false"
                                aria-label="Select 1"
                                class="p-checkbox-box p-component"
                                role="checkbox"
                                tabindex="0"
                              >
                                <span
                                  class="p-checkbox-icon"
                                />
                              </div>
                            </div>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="nameTitle"
                              >
                                Name
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              Item 1
                            </span>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="sizeTitle"
                              >
                                Size
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              10
                            </span>
                          </td>
                        </tr>
                        <tr
                          class="p-selectable-row"
                          role="row"
                        >
                          <td
                            class="flex-grow-0 p-selection-column"
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            />
                            <div
                              class="p-checkbox p-component"
                            >
                              <div
                                aria-checked="false"
                                aria-label="Select 2"
                                class="p-checkbox-box p-component"
                                role="checkbox"
                                tabindex="0"
                              >
                                <span
                                  class="p-checkbox-icon"
                                />
                              </div>
                            </div>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="nameTitle"
                              >
                                Name
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              Item 2
                            </span>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="sizeTitle"
                              >
                                Size
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              20
                            </span>
                          </td>
                        </tr>
                        <tr
                          class="p-selectable-row p-row-odd"
                          role="row"
                        >
                          <td
                            class="flex-grow-0 p-selection-column"
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            />
                            <div
                              class="p-checkbox p-component"
                            >
                              <div
                                aria-checked="false"
                                aria-label="Select 3"
                                class="p-checkbox-box p-component"
                                role="checkbox"
                                tabindex="0"
                              >
                                <span
                                  class="p-checkbox-icon"
                                />
                              </div>
                            </div>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="nameTitle"
                              >
                                Name
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              Item 3
                            </span>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="sizeTitle"
                              >
                                Size
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              30
                            </span>
                          </td>
                        </tr>
                        <tr
                          class="p-selectable-row"
                          role="row"
                        >
                          <td
                            class="flex-grow-0 p-selection-column"
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            />
                            <div
                              class="p-checkbox p-component"
                            >
                              <div
                                aria-checked="false"
                                aria-label="Select 4"
                                class="p-checkbox-box p-component"
                                role="checkbox"
                                tabindex="0"
                              >
                                <span
                                  class="p-checkbox-icon"
                                />
                              </div>
                            </div>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="nameTitle"
                              >
                                Name
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              Item 4
                            </span>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="sizeTitle"
                              >
                                Size
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              40
                            </span>
                          </td>
                        </tr>
                        <tr
                          class="p-selectable-row p-row-odd"
                          role="row"
                        >
                          <td
                            class="flex-grow-0 p-selection-column"
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            />
                            <div
                              class="p-checkbox p-component"
                            >
                              <div
                                aria-checked="false"
                                aria-label="Select 5"
                                class="p-checkbox-box p-component"
                                role="checkbox"
                                tabindex="0"
                              >
                                <span
                                  class="p-checkbox-icon"
                                />
                              </div>
                            </div>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="nameTitle"
                              >
                                Name
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              Item 5
                            </span>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="sizeTitle"
                              >
                                Size
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              50
                            </span>
                          </td>
                        </tr>
                        <tr
                          class="p-selectable-row"
                          role="row"
                        >
                          <td
                            class="flex-grow-0 p-selection-column"
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            />
                            <div
                              class="p-checkbox p-component"
                            >
                              <div
                                aria-checked="false"
                                aria-label="Select 6"
                                class="p-checkbox-box p-component"
                                role="checkbox"
                                tabindex="0"
                              >
                                <span
                                  class="p-checkbox-icon"
                                />
                              </div>
                            </div>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="nameTitle"
                              >
                                Name
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              Item 6
                            </span>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="sizeTitle"
                              >
                                Size
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              60
                            </span>
                          </td>
                        </tr>
                        <tr
                          class="p-selectable-row p-row-odd"
                          role="row"
                        >
                          <td
                            class="flex-grow-0 p-selection-column"
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            />
                            <div
                              class="p-checkbox p-component"
                            >
                              <div
                                aria-checked="false"
                                aria-label="Select 7"
                                class="p-checkbox-box p-component"
                                role="checkbox"
                                tabindex="0"
                              >
                                <span
                                  class="p-checkbox-icon"
                                />
                              </div>
                            </div>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="nameTitle"
                              >
                                Name
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              Item 7
                            </span>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="sizeTitle"
                              >
                                Size
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              70
                            </span>
                          </td>
                        </tr>
                        <tr
                          class="p-selectable-row"
                          role="row"
                        >
                          <td
                            class="flex-grow-0 p-selection-column"
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            />
                            <div
                              class="p-checkbox p-component"
                            >
                              <div
                                aria-checked="false"
                                aria-label="Select 8"
                                class="p-checkbox-box p-component"
                                role="checkbox"
                                tabindex="0"
                              >
                                <span
                                  class="p-checkbox-icon"
                                />
                              </div>
                            </div>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="nameTitle"
                              >
                                Name
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              Item 8
                            </span>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="sizeTitle"
                              >
                                Size
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              80
                            </span>
                          </td>
                        </tr>
                        <tr
                          class="p-selectable-row p-row-odd"
                          role="row"
                        >
                          <td
                            class="flex-grow-0 p-selection-column"
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            />
                            <div
                              class="p-checkbox p-component"
                            >
                              <div
                                aria-checked="false"
                                aria-label="Select 9"
                                class="p-checkbox-box p-component"
                                role="checkbox"
                                tabindex="0"
                              >
                                <span
                                  class="p-checkbox-icon"
                                />
                              </div>
                            </div>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="nameTitle"
                              >
                                Name
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              Item 9
                            </span>
                          </td>
                          <td
                            class=""
                            role="cell"
                          >
                            <span
                              class="p-column-title"
                            >
                              <span
                                data-testid="sizeTitle"
                              >
                                Size
                              </span>
                            </span>
                            <span
                              class="value"
                            >
                              90
                            </span>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                  <div
                    class="p-paginator p-component p-paginator-bottom"
                  >
                    <button
                      aria-label="First Page"
                      class="p-paginator-first p-paginator-element p-link p-disabled"
                      disabled=""
                      type="button"
                    >
                      <span
                        class="p-paginator-icon pi pi-angle-double-left"
                      />
                    </button>
                    <button
                      aria-label="Previous Page"
                      class="p-paginator-prev p-paginator-element p-link p-disabled"
                      disabled=""
                      type="button"
                    >
                      <span
                        class="p-paginator-icon pi pi-angle-left"
                      />
                    </button>
                    <span
                      class="p-paginator-pages"
                    >
                      <button
                        aria-label="Page 2"
                        class="p-paginator-page p-paginator-element p-link p-paginator-page-start p-highlight"
                        type="button"
                      >
                        1
                      </button>
                      <button
                        aria-label="Page 3"
                        class="p-paginator-page p-paginator-element p-link"
                        type="button"
                      >
                        2
                      </button>
                      <button
                        aria-label="Page 4"
                        class="p-paginator-page p-paginator-element p-link"
                        type="button"
                      >
                        3
                      </button>
                      <button
                        aria-label="Page 5"
                        class="p-paginator-page p-paginator-element p-link"
                        type="button"
                      >
                        4
                      </button>
                      <button
                        aria-label="Page 6"
                        class="p-paginator-page p-paginator-element p-link p-paginator-page-end"
                        type="button"
                      >
                        5
                      </button>
                    </span>
                    <button
                      aria-label="Next Page"
                      class="p-paginator-next p-paginator-element p-link"
                      type="button"
                    >
                      <span
                        class="p-paginator-icon pi pi-angle-right"
                      />
                    </button>
                    <button
                      aria-label="Last Page"
                      class="p-paginator-last p-paginator-element p-link"
                      type="button"
                    >
                      <span
                        class="p-paginator-icon pi pi-angle-double-right"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="p-splitter-gutter"
              style="width: 4px;"
            >
              <div
                class="p-splitter-gutter-handle"
              />
            </div>
            <div
              class="p-splitter-panel"
              style="flex-basis: calc(10% - 4px); height: 768px;"
            >
              <div
                class="w-full"
              >
                <div>
                  <div>
                    Name
                    : 
                  </div>
                  <div>
                    Size
                    : 
                  </div>
                  <div>
                    Records
                    : 
                    55
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
